fix(docs) added troubleshooting section for iOS XCFrameworks on EAS (#46287)
expo/expo: docs/pages/guides/prebuilt-expo-modules.mdx

This is typically only needed when you're modifying module source code yourself.

</Collapsible>

<Collapsible summary="Troubleshooting EAS Build">

### iOS

> On EAS Build, third-party libraries like `react-native-reanimated` and `react-native-worklets` are automatically downloaded as precompiled XCFrameworks. Local `pod install` does not fetch them by default — those packages build from source locally and pick up any `staticFeatureFlags` overrides automatically — so flag and version mismatches surface primarily on EAS Build. Enabling third-party precompiled downloads for local builds is not recommended; keep this path scoped to EAS.

Reviewer: **[suggestion]** Long sentence with two em-dash insertions, plus a passive trailing clause (`expo-docs-style-guide.md#punctuation`, `expo-docs-style-guide.md#voice-and-tone`). Prefer separate sentences for phrase splitting and active voice for the recommendation.

```suggestion
On EAS Build, third-party libraries like `react-native-reanimated` and `react-native-worklets` are automatically downloaded as precompiled XCFrameworks. Local `pod install` does not fetch them by default. Those packages build from source locally and pick up any `staticFeatureFlags` overrides automatically, so flag and version mismatches surface primarily on EAS Build. Avoid enabling third-party precompiled downloads for local builds and keep this path scoped to EAS.
```

#### `react-native-reanimated` and `react-native-worklets`

> These two packages are tightly coupled — `react-native-reanimated` links `react-native-worklets` at the native level.

Reviewer: **[suggestion]** Em dash splits two complete thoughts (`expo-docs-style-guide.md#punctuation`). Use a period.

```suggestion
These two packages are tightly coupled. `react-native-reanimated` links `react-native-worklets` at the native level.
```

**Opt out of both together.** If you need a source build for either package (for example to apply a patch or modify native code), list both in `buildFromSource`. Source-building only one produces a mixed precompiled/source linkage that fails to resolve the matching framework at runtime:

{/* prettier-ignore */}
```json package.json
{
  "expo": {
    "autolinking": {
      "ios": {
        "buildFromSource": ["react-native-reanimated", "react-native-worklets"]
      }
    }
  }
}
```

> **Custom feature flags require a source build.** Feature-flag values are baked into the precompiled binary at build time, so any `worklets.staticFeatureFlags` or `reanimated.staticFeatureFlags` overrides in **package.json** are ignored. To apply them, disable precompile with `EXPO_USE_PRECOMPILED_MODULES=0`.

Reviewer: **[nit]** "disable precompile" introduces a bare-noun form that doesn't appear elsewhere on the page — every other reference uses "precompiled" as an adjective (`expo-docs-style-guide.md#glossary-and-product-names`). Keep the term consistent.

```suggestion
**Custom feature flags require a source build.** Feature-flag values are baked into the precompiled binary at build time, so any `worklets.staticFeatureFlags` or `reanimated.staticFeatureFlags` overrides in **package.json** are ignored. To apply them, disable precompiled modules with `EXPO_USE_PRECOMPILED_MODULES=0`.
```

> **When to look here.** Runtime errors like `Unable to recognize flag: <NAME>` on EAS Build (but not locally) mean the precompiled artifact's flag list doesn't match your pinned package version — use `buildFromSource` above and file an issue.

Reviewer: **[design]** "file an issue" has no destination, and the em dash mid-sentence separates the diagnosis from the action (`expo-docs-style-guide.md#formatting-and-structure`, `expo-docs-style-guide.md#punctuation`). The reader is mid-incident on a troubleshooting page; link the action and split the sentence.

```suggestion
**When to look here.** Runtime errors like `Unable to recognize flag: <NAME>` on EAS Build (but not locally) mean the precompiled artifact's flag list doesn't match your pinned package version. Use `buildFromSource` above and [file an issue on GitHub](https://github.com/expo/expo/issues).
```
